framework/validators/IpValidator.php

```diff
@@ -0,0 +1,555 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+use yii\helpers\Json;
+use yii\helpers\StringHelper;
+use yii\web\JsExpression;
+
+/**
+ * IpValidator validates that the attribute value is a valid IPv4/IPv6 address or subnet.
+ * May change attribute's value if normalization is enabled.
+ *
+ * @property boolean|string negationChar whether address may have an negation-character at the beginning
+ *   string - passed character will be used. May be a regular expression
+ *   true   - character from [[DEFAULT_NEGATION_CHAR]] (default `!`) will be used
+ *   false  - negation is not allowed
+ * @author SilverFire <d.naumenko.a@gmail.com>
+ * @since 2.0.5
+ */
+class IpValidator extends Validator
+{
+    /**
+     * @const integer the length of IPv6 address in bits
+     */
+    const IPV6_ADDRESS_LENGTH = 128;
+
+    /**
+     * @const integer the length of IPv4 address in bits
+     */
+    const IPV4_ADDRESS_LENGTH = 32;
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed()
+     * @see order
+     */
+    const ORDER_ALLOW_DENY = 0;
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed()
+     * @see order
+     */
+    const ORDER_DENY_ALLOW = 1;
+
+    /**
+     * @const string default negation char when [[negationChar]] is set to `true`
+     * @see negationChar
+     */
+    const DEFAULT_NEGATION_CHAR = '!';
+
+    /**
+     * @var boolean whether support of IPv6 addresses is enabled
+     */
+    public $ipv6 = true;
+
+    /**
+     * @var boolean whether support of IPv4 addresses is enabled
+     */
+    public $ipv4 = true;
+
+    /**
+     * @var boolean whether the address can be a CIDR subnet
+     *    true - the subnet is required
+     *   false - the address can not have the subnet
+     *    null - ths subnet is optional
+     */
+    public $subnet = false;
+
+    /**
+     * @var boolean whether to add the prefix with the smallest length (32 for IPv4 and 128 for IPv6)
+     * to an address without it.
+     * Works only when attribute [[subnet]] is not false.
+     * Example: `10.0.1.5` will normalized to `10.0.1.5/32`, `2008:db0::1` will be normalized to `2008:db0::1/128`
+     * @see subnet
+     */
+    public $normalize = false;
+
+    /**
+     * @var string|boolean
+     * @see negationChar
+     */
+    public $_negationChar = false;
+
+    /**
+     * @var boolean whether to expand an IPv6 address to the full notation format
+     */
+    public $expandV6 = false;
```

Comment:
Yes, I have also thought about this, but wanted to leave it a little bit shorter